### ISSUE TITLE
bug: `use_context` doesn't fetch resource at the top level (`SsrMode::PartiallyBlocked`)

### ISSUE THREAD

**Issue by Skyliegirl33** [leptos-rs/leptos#1095]
Given some code like this, where `provide_auth_context(cx)` provides a struct with a resource:
```rust
#[component]
pub fn App(cx: Scope) -> impl IntoView {
  provide_meta_context(cx);
  provide_auth_context(cx);
  let auth_context = use_context::<AuthContext>(cx).expect("Failed to find AuthContext");
  
  view! {
    cx,
    <Routes>
      <Route path="auth/settings"
        view=move |cx| view! { cx,
        <Transition fallback=|| ()>
          {move || {
            match auth_context.user.read(cx) {
              Some(Ok(Some(_))) => view! { cx, <h1>"Settings"</h1> }.into_view(cx),
              None => ().into_view(cx),
              _ => view! { cx, <Redirect path="/"/> }.into_view(cx)
            }
          }}
        </Transition>
        }
        ssr=SsrMode::PartiallyBlocked
      />
    </Routes>
  }
}
```
makes the resource not load properly.
The expected behavior would be that the resource resolves on the server and returns `<h1>"Settings"</h1>`
Note that it doesn't resolve on the client either until I navigate to another page and then go back.

The problem can be mitigated by putting `use_context` inside the `view!` of the Route

**Comment by gbj** [comment]
I wonder whether this is fixed after I just merged PR #1110. It fixed it in my attempted reproduction but could you double check it in the example this bug came up in for you? If not fixed, remind me of a link to find it and I'll try to fix this one against your actual example instead of my repro.

**Comment by Skyliegirl33** [comment]
It does seem to be resolved now. Thanks!
